src/cpp/include/openvino/genai/streamer_base.hpp

```diff
@@ -8,23 +8,34 @@
 namespace ov {
 namespace genai {
 
-/** 
+/**
  * @brief base class for streamers. In order to use inherit from from this class and implement put, and methods
- * 
+ *
  * @param m_tokenizer tokenizer
-*/
+ */
 class StreamerBase {
 public:
     /// @brief put is called every time new token is decoded,
     /// @return bool flag to indicate whether generation should be stopped, if return true generation stops
     virtual bool put(int64_t token) = 0;
-    
+
     /// @brief end is called at the end of generation. It can be used to flush cache if your own streamer has one
     virtual void end() = 0;
 
     virtual ~StreamerBase() = default;
```

Comment:
it's better to move dtor definition to .cpp file and export this class

it will help with RTTI issue on some platforms

Reply:
Done. I hope I understood it correctly, could you please double check.